### ISSUE TITLE
Use of MediaWiki\Parser\Parser::getOutput before initialization was deprecated in MediaWiki 1.42

### ISSUE THREAD

**Issue by paladox** [SemanticMediaWiki/SemanticMediaWiki#5872]
> Deprecated: Use of MediaWiki\Parser\Parser::getOutput before initialization was deprecated in MediaWiki 1.42. [Called from SMW\Parser\RecursiveTextProcessor::getParserOutputSafe in /var/www/html/extensions/SemanticResultFormats/extensions/SemanticMediaWiki/src/Parser/RecursiveTextProcessor.php at line 298] in /var/www/html/includes/debug/MWDebug.php on line 379

**Comment by paladox** [comment]
> So, this issue should belong to the SemanticResultFormats repository, right?

I don’t think so? Not sure tho, it’s coming from SMW.

**Comment by jaideraf** [comment]
Oh, this extract mislead me:

`/var/www/html/extensions/SemanticResultFormats/`

**Comment by YOUR1** [comment]
I think this should be a part of the 5.0.0 milestone, since it's visible for every user.

**Comment by YOUR1** [comment]
This notice also displayes when the user searches for something on Special:FacettedSearch. 

```
Deprecated: Use of MediaWiki\Parser\Parser::getOutput before initialization was deprecated in MediaWiki 1.42. [Called from SMW\Parser\RecursiveTextProcessor::getParserOutputSafe in /var/platform/mediawiki/1.43/core/20250127133002/extensions/SemanticMediaWiki/src/Parser/RecursiveTextProcessor.php at line 298] in /var/platform/mediawiki/1.43/core/20250127133002/includes/debug/MWDebug.php on line 385
```

**Comment by jaideraf** [comment]
I couldn't reproduce here: https://wikincat.org/w/index.php?title=Especial:FacetedSearch&q=%5B%5BCategory%3AObra%5D%5D&profile=work

**Comment by Sophivorus** [comment]
This deprecation notice is currently our top log spam

**Comment by YOUR1** [comment]
> I couldn't reproduce here: https://wikincat.org/w/index.php?title=Especial:FacetedSearch&q=%5B%5BCategory%3AObra%5D%5D&profile=work

Does it have error reporting on?

> This deprecation notice is currently our top log spam, so we would appreciate a fix before the 5.0 milestone.
> 
> ![Image](https://github.com/user-attachments/assets/c08092ba-b402-4234-8ddb-1198d912ab6b)

Same here.

**Comment by YOUR1** [comment]
Im able to reproduce this constantly on the current master version of SMW + MW 1.43.

The way I reproduced it was as follow:
* Create page `Property:Username` with `[[Has type::String]]`
* Create page `James` with `[[Username::mwjames]]`
* Run jobs
* Go to `Special:FacetedSearch` and search for `[[Username::+]]`


`Deprecated: Use of MediaWiki\Parser\Parser::getOutput before initialization was deprecated in MediaWiki 1.42. [Called from SMW\Parser\RecursiveTextProcessor::getParserOutputSafe in /var/www/html/extensions/SemanticMediaWiki/src/Parser/RecursiveTextProcessor.php at line 298] in /var/www/html/includes/debug/MWDebug.php on line 385`

**Comment by jaideraf** [comment]
> Does it have error reporting on?

Yes. I have 

```php
$wgShowExceptionDetails = true;
$wgDevelopmentWarnings = true;
```

```ini
error_reporting = E_ALL
display_errors = On
```

Yet, I can't reproduce the error: https://wikincat.org/wiki/Especial:FacetedSearch?profile=default&q=%5B%5BUsername%3A%3A%2B%5D%5D&utf8=%E2%9C%93&limit=50&offset=0&csum=464215950&pv%5BUsername%5D%5B%5D=&pv%5BUsername%5D%5B%5D=jaideraf&format=table&order=asc&size=50

Notice: `[[Has type::String]]` does not exist anymore, the correct form is  `[[Has type::Text]]`.

**Comment by jaideraf** [comment]
> Hmm, also with the exact steps I mentioned before?

Yes, I only replaced MWJames user to my user, as you can see in the above link.

**Comment by YOUR1** [comment]
Weird. Did you check the docker setup as well? I can reproduce it every time with a new install..

**Comment by paladox** [comment]
Do you use a job runner or something? I think it only happens if something wasn’t parsed before it is hit? I saw this on cargo and fixed it by doing https://github.com/wikimedia/mediawiki-extensions-Cargo/commit/061fb700e7fd54ff8028a48270d9fea9f658e4c5

**Comment by YOUR1** [comment]
I might had `$wgJobRunRate` set to 0.
